packages/host/app/services/command-service.ts

```diff
@@ -0,0 +1,68 @@
+import Service from '@ember/service';
+
+type Command = 'patch';
+
+class CommandHandler {
+  handler: any;
+  functions: Map<Command, ((arg: any) => void)[]>;
+  constructor(handler: any) {
+    this.handler = handler;
+    this.functions = new Map();
+  }
+
+  public get(command: Command) {
+    if (!this.functions.has(command)) {
+      this.functions.set(command, []);
+    }
+    return this.functions.get(command);
+  }
+
+  public async handle(command: Command, arg: any) {
+    if (!this.functions.has(command)) {
+      throw new Error(`No handler registered for command ${command}`);
+    }
+    const functions = this.functions.get(command)!;
+    for (const func of functions) {
+      await func(arg);
+    }
+  }
+  public setHandler(command: Command, func: (arg: any) => void) {
+    if (!this.functions.has(command)) {
+      this.functions.set(command, []);
+    }
+    this.functions.get(command)!.push(func);
+  }
+}
+
+export default class CommandService extends Service {
+  commandHandlers: Map<any, CommandHandler>;
+
+  constructor(properties: object) {
+    super(properties);
+    this.commandHandlers = new Map();
```

Comment:
no need to have a constructor override just to initialize a property, just include
```
private commandHandlers = new Map();
```
that is the same as setting it in the constructor--then we won't need a constructor override.